packages/ui/src/views/repo/repo-summary/index.tsx

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { useNavigate } from 'react-router-dom'
+
+import {
+  BranchSelector,
+  Button,
+  ButtonGroup,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  ListActions,
+  NoData,
+  RepoSummaryPanel,
+  SearchFiles,
+  SkeletonList,
+  Spacer,
+  StackedList,
+  Summary,
+  Text,
+  type BranchSelectorItem
+} from '@/components'
+
+import { SandboxLayout } from '../../index'
+
+interface RepoSummaryViewProps {
+  loading: boolean
+  selectedBranch: string
+  branchList: BranchSelectorItem[]
+  tagList?: BranchSelectorItem[]
+  selectBranch: (branch: string) => void
+  filesList: string[]
+  navigateToFile: (path: string) => void
+  repository:
+    | {
+        git_ssh_url?: string
+        git_url?: string
+        description?: string
+      }
+    | undefined
+  handleCreateToken: () => void
+  // TODO: fix this
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  repoEntryPathToFileTypeMap: Map<string, any>
+  // TODO: fix this
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  files: any[]
+  decodedReadmeContent: string
+  summaryDetails: {
+    default_branch_commit_count?: number
+    branch_count?: number
+    tag_count?: number
+    pull_req_summary?: { open_count: number } | undefined
+  }
+  spaceId: string
+  repoId: string
+  gitRef?: string
+  latestCommitInfo?: {
+    userName: string
+    message: string
+    timestamp: string
+    sha: string | null
+  }
+}
+
+export function RepoSummaryView({
+  loading,
+  selectedBranch,
+  branchList,
+  tagList = [],
+  selectBranch,
+  filesList,
+  navigateToFile,
+  repository,
+  handleCreateToken,
+  repoEntryPathToFileTypeMap,
+  files,
+  decodedReadmeContent,
+  summaryDetails: { default_branch_commit_count = 0, branch_count = 0, tag_count = 0, pull_req_summary },
+  spaceId,
+  repoId,
+  gitRef,
+  latestCommitInfo
+}: RepoSummaryViewProps) {
+  const navigate = useNavigate()
+
+  const renderListContent = () => {
+    if (loading) return <SkeletonList />
+
+    if (!loading && repoEntryPathToFileTypeMap.size > 0) {
```

Comment:
No need to use `> 0` or check for `!loading`
```suggestion
    if (repoEntryPathToFileTypeMap.size) {
```